app/controllers/mixins/generic_show_mixin.rb

```diff
@@ -107,6 +107,26 @@ def display_nested_generic(display)
       nested_list(display, display.camelize.singularize.constantize)
     end
 
+    def display_descendant_vms
+      @showtype = "config"
+      drop_breadcrumb(:name => _("%{name} (All VMs - Tree View)") % {:name => @record.name},
+                      :url  => "/#{controller_name}/show/#{@record.id}?display=descendant_vms&treestate=true")
```

Comment:
good point with ```show_link```, i'm going to use that. For ```descendant_vms``` there will be at least one more other controller (```ems_cluster```) that's going to use the method.